outbreaks/views.py

```
@@ -84,130 +103,147 @@ def get_context_data(self, *args, **kwargs):
-                if num_dates > 1:
-                    num_dates -= 1
-                    self.request.session["num_dates"] = num_dates
-                    self.request.session.pop(f"alert_datetime_{num_dates}", None)
-                return redirect(
-                    reverse_lazy("outbreaks:datetime") + f"?num_dates={num_dates}"
-                )
-
-        # Proceed to success or form error
-        return response
+        """
+        POST scenarios:
+        * Add time range -> display date fields
+        * Clear time range -> hide date fields and reset to default start / end
+        * Add date (save to session) -> validate form and save
+        * Show add date -> display date form
+        * Unshow add date -> hide date form
+        * Remove date
+        * Done/Next (no more to save) -> redirect to success
+        """
+        show_time_fields = False
+        show_date_form = True
+        show_errors = False
+        success_url = reverse_lazy("outbreaks:severity")
+        do_post = request.POST.get("do_post", None)
+        post_data = request.POST
+        next_button_show = not request.session.get('selected_dates')
+        next_button_submit_form_ind = not request.session.get('selected_dates')
+        form = None
+        if not do_post:  # no form to submit, going next
+            if not request.session.get('selected_dates'):
+                return HttpResponseRedirect(reverse_lazy('outbreaks:datetime'))
+            return HttpResponseRedirect(success_url)
+
+        if do_post == 'add_date':  # submitting a form
+            form = DateForm(
+                post_data,
+                alert_location=request.session['alert_location'],
+                show_time_fields=post_data.get("start_time", None)
+            )
+            if form.is_valid():
+                if self.save_form_dates(form):
+                    return HttpResponseRedirect(reverse_lazy('outbreaks:datetime'))
+            show_errors = True
```

this is used to hide errors on the form when not really submitting it by user intent